File: leet_code_algorithms/two_sum/practice/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

/*My Answer*/
var twoSum = function (nums, target) {
  let seen = [];
  let indices = [];
  nums.forEach((num, i) => {
    if(seen.includes(target-num)) {
      indices = [i, seen.indexOf(target-num)]
    } else {
      seen.push(num)
    }
  })
  return indices;
}

var twoSum = function (nums, target) {
  let map = new Map();
  for (let i = 0; i < nums.length; i++) {
    let complement = target - nums[i];
    if(map.has(complement)) {
      return [map.get(complement), i];
    } else {
      map.set(nums[i], i);
    }
  }
  return []
}

File: leet_code_algorithms/two_sum/practice/two_sum.py
from typing import List

def twoSum(self, nums: List[int], target: int) -> List[int]:
    seen = [nums[0]]
    for num in range(1, len(nums)):
        diff = target - nums[num]
        if diff in seen:
            return [seen.index(diff), num]
        else:
            seen.append(nums[num])

print(twoSum("",[2,7,11,15],9))